astroquery/vo_conesearch/async.py

```diff
@@ -0,0 +1,80 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Asynchronous VO service requests."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+# ASTROPY
+from astropy.utils.compat.futures import ThreadPoolExecutor
+
+__all__ = ['AsyncBase']
+
+
+class AsyncBase(object):
+    """Base class for asynchronous VO service requests
+    using :py:class:`concurrent.futures.ThreadPoolExecutor`.
+
+    Service request will be forced to run in silent
+    mode by setting ``verbose=False``. Warnings are controlled
+    by :py:mod:`warnings` module.
+
+    .. note::
+
+        Methods of the attributes can be accessed directly,
+        with priority given to ``executor``.
+
+    Parameters
+    ----------
+    func : function
+        The function to run.
+
+    args, kwargs
+        Arguments and keywords accepted by the service request
+        function to be called asynchronously.
+
+    Attributes
+    ----------
+    executor : :py:class:`concurrent.futures.ThreadPoolExecutor`
+        Executor running the function on single thread.
+
+    future : :py:class:`concurrent.futures.Future`
+        Asynchronous execution created by ``executor``.
+
+    """
+    def __init__(self, func, *args, **kwargs):
+        kwargs['verbose'] = False
+        self.executor = ThreadPoolExecutor(1)
+        self.future = self.executor.submit(func, *args, **kwargs)
+
+    def __getattr__(self, what):
+        """Expose ``executor`` and ``future`` methods."""
+        try:
+            return getattr(self.executor, what)
```

Comment:
Just throwing in some random stuff: `getattr` takes a third argument in case it encounters an `AttributeError`. However `return getattr(self.executor, what, getattr(self.future, what))` might be hard to swallow 😄